packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -292,6 +334,37 @@ describe('<DataGrid /> - Pagination', () => {
       fireEvent.click(screen.queryAllByRole('option')[1]);
       expect(getColumnValues()).to.deep.equal(['0', '1']);
     });
+
+    it('should display a warning if the prop pageSize is not in the prop rowsPerPageOptions', () => {
+      const pageSize = 12;
+
+      expect(() => {
+        render(<BaselineTestCase pageSize={pageSize} rowsPerPageOptions={[25, 50, 100]} />);
+        // @ts-expect-error need to migrate helpers to TypeScript
+      }).toWarnDev([
+        `Material-UI: The page size \`${pageSize}\` is not preset in the \`rowsPerPageOptions\``,
+      ]);
+    });
+
+    it('should display a warning if the prop pageSize is not in the default rowsPerPageOptions', () => {
+      const pageSize = 12;
+
+      expect(() => {
+        render(<BaselineTestCase pageSize={pageSize} />);
+        // @ts-expect-error need to migrate helpers to TypeScript
+      }).toWarnDev([
+        `Material-UI: The page size \`${pageSize}\` is not preset in the \`rowsPerPageOptions\``,
+      ]);
+    });
+
+    it('should display a warning if the default pageSize given as props is not in the prop rowsPerPageOptions', () => {
+      expect(() => {
+        render(<BaselineTestCase rowsPerPageOptions={[25, 50]} />);
+        // @ts-expect-error need to migrate helpers to TypeScript
+      }).toWarnDev([
+        `Material-UI: The page size \`${DEFAULT_PAGE_SIZE}\` is not preset in the \`rowsPerPageOptions\``,
```

Comment:
I wouldn't mind hardcoding the value if we want to avoid exporting constants.

```suggestion
        `Material-UI: The page size \`100\` is not preset in the \`rowsPerPageOptions\``,
```